system/lib/libcxxabi/src/cxa_exception_emscripten.cpp

```diff
@@ -50,6 +47,9 @@ thrown_object_from_cxa_exception(__cxa_exception* exception_header) {
   return static_cast<void*>(exception_header + 1);
 }
 
+#if defined(__USING_EMSCRIPTEN_EXCEPTIONS__) ||                                \
+  defined(__USING_WASM_EXCEPTIONS__)
+
```

Comment:
Why move this? The functions above this and below this are similar utility functions.

Reply:
These two functions are needed by __cxa_allocate_exception, __cxa_free_exception and those functions are included even when `__USING_EMSCRIPTEN_EXCEPTIONS__` is not defined.  

`__USING_EMSCRIPTEN_EXCEPTIONS__` is defined based on `DISABLE_EXCEPTION_CATCHING` and not based on `DISABLE_EXCEPTION_CATCHING`

Reply:
i.e. we need __cxa_allocate_exception and __cxa_free_exception even when neither `__USING_EMSCRIPTEN_EXCEPTIONS__` nor `__USING_WASM_EXCEPTIONS__` is defined.  In this mode exceptions can still be thrown.. they just can't be caught.